### PR TITLE
Refactor

### LINKED ISSUE #79: The ability to encode lyrics into the song file, instead of saving a .lrc file
I am frustrated when I have to save a .lrc file instead of embedding it into the actual song file, therefore adding more visual clutter to my file system and making sure I have to transfer more files when I move my music library.

**Describe the solution you'd like**
The solution would be to have an option to imbed the lyrics into the song file instead of saving them as a lyric file.

**Describe alternatives you've considered**
 The only alternative is the current behavior which is to save a lyric file

**Additional context**
Other apps have this option but do not even search for synced lyrics in the first place

### PR DESCRIPTION
**Pull request by nxoim**
- Refactored a little hehe
- Improved fab -> lyrics fetcher transition
- improved some lyrics fetcher ui

Closes #79

### CODE REVIEW

**Comment by nxoim** [comment]
please squash it on merge

File: app/src/main/java/pl/lambada/songsync/data/remote/lyrics_providers/LyricsProviderService.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    /**
+     * Gets synced lyrics using the song link and returns them as a string formatted as an LRC file.
+     * @param songLink The link to the song.
+     * @return The synced lyrics as a string.
+     */
+    suspend fun getSyncedLyrics(
+        songLink: String,
+        version: String,
+        provider: Providers,
+        // TODO providers could be a sealed interface to include such parameters
```
Comment (BobbyESP) [inline app/src/main/java/pl/lambada/songsync/data/remote/lyrics_providers/LyricsProviderService.kt:87]: True, but let's leave this to another PR

Comment (nxoim) [reply app/src/main/java/pl/lambada/songsync/data/remote/lyrics_providers/LyricsProviderService.kt:87]: feel free to merge if you dont find any other issues

**Review by Lambada10 — APPROVED** [review]
